### PR TITLE
feat(nodejs): ipc & parquet support

### LINKED ISSUE #2400: Add IPC & better Parquet support to node api
Arrow and Parquet read/write methods should have feature parity with python & rust

### PR DESCRIPTION
**Pull request by universalmind303**
Right now, this only supports writing to a file path for both IPC & parquet. Unlike CSV & JSON,  the underlying writers require `Seek` to be implemented, which can't be done for a node stream. 

closes #2400

### CODE REVIEW

**Comment by ritchie46** [comment]
For reading you could always use file paths right?

**Comment by universalmind303** [comment]
> For reading you could always use file paths right?

could you rephrase the question. I'm not sure i understand.

File: nodejs-polars/polars/io.ts [hunk 1]
```diff
@@ -181,20 +499,24 @@ class LineBatcher extends Stream.Transform {
...
-    for (var i = 0; i < chunk.length; i++) {
+    var position = 0;
+
+    let i = 0;
+    while (i < chunk.length) {
       if (chunk[i] === 10) { // '\n'
         this.#accumulatedLines++;
         if (this.#accumulatedLines == this.#batchSize) {
-          this.#lines.push(chunk.slice(begin, i + 1));
+          this.#lines.push(chunk.subarray(begin, i + 1));
```
Comment (universalmind303) [inline nodejs-polars/polars/io.ts:509]: subarray gets the chunk by reference instead of copy.

File: nodejs-polars/polars/io.ts [hunk 2]
```diff
@@ -181,20 +499,24 @@ class LineBatcher extends Stream.Transform {
   _transform(chunk, _encoding, done) {
 
     var begin = 0;
-    for (var i = 0; i < chunk.length; i++) {
+    var position = 0;
+
+    let i = 0;
+    while (i < chunk.length) {
```
Comment (universalmind303) [inline nodejs-polars/polars/io.ts:505]: while loops are slightly faster than for loops
